runtime/reconcilers/model.go

```diff
@@ -598,6 +610,9 @@ func (r *ModelReconciler) resolveIncrementalState(ctx context.Context, mdl *runt
 // acquireExecutor acquires a ModelExecutor capable of executing a model with the given execution options.
 // It handles acquiring and setting opts.InputHandle and opts.OutputHandle.
 func (r *ModelReconciler) acquireExecutor(ctx context.Context, opts *drivers.ModelExecutorOptions) (string, drivers.ModelExecutor, func(), error) {
+	if opts.StageConnector != "" {
+		return r.acquireExecutorWithStage(ctx, opts)
+	}
```

Comment:
It would be nice to break this out from `acquireExecutor` because `acquireExecutorWithStage` calls `acquireExecutor`, which makes the logic more convoluted than necessary.